openroad_interface/exec.py

```diff
@@ -0,0 +1,76 @@
+
+def exec(design_file, test_file):
+    from var import directory
+    from def_generator import def_generator
+    import networkx as nx
+    import os
+    import shutil
+    import parasitic_estimation as pe
+    from length_calculations import length_calculations
+    import parasitic_calc as pc
+
+    # 1. generate def 
+    os.system("export PATH=./OpenROAD/build/src:$PATH") ## please figure this out 
+    os.system("cp " + test_file + " ./" + directory) 
+    shutil.copyfile(test_file, directory + "test.tcl")
+
+    graph, net_out_dict, output_dict, lef_pitch, layer_res, layer_cap, units = def_generator(test_file, "test_files/" + design_file + ".gml")
+
+    # 2. run openroad
+    os.chdir(directory)
+    os.system("openroad test.tcl")
+    os.chdir("../..")
+
+    # 3. run parasitic_calc and length_calculations
+    net_cap, net_res = pc.parasitic_calc()
+    length_list = length_calculations()
+
+    # 4. add edge attributions
+    res_graph_data = []
+    cap_graph_data = []
+    len_graph_data = []
+    print (length_list)
+    print(net_out_dict)
+    for output_pin in net_out_dict:
+        for pin in output_dict[output_pin]:
+            graph[output_pin][pin]['net'] = net_out_dict[output_pin]
+            graph[output_pin][pin]['net_length'] = length_list[net_out_dict[output_pin]]
+            graph[output_pin][pin]['net_res'] = net_res[net_out_dict[output_pin]][3]
+            graph[output_pin][pin]['net_cap'] = net_cap[net_out_dict[output_pin]]
+        res_graph_data.append(float(net_res[net_out_dict[output_pin]][3]))
+        cap_graph_data.append(float(net_cap[net_out_dict[output_pin]])* pow(10,4))
+        len_graph_data.append(float(length_list[net_out_dict[output_pin]]))
+
```

Comment:
All this should be in a separate function that takes a nx.Digraph as input, and returns a separate nx.Digraph with all the edge attributes.

Reply:
call the file `place_n_route.py`